WordPress/gutenberg
# Flex: Remove flex `gap` polyfill

## What?

Removes the margin-based polyfill implementation of flex `gap`.

## Why?

Flex `gap` is now at an acceptable [adoption](https://caniuse.com/mdn-css_properties_gap_flex_context) percentage.

Using the browser's native flex gap will make our code much simpler, and will avoid complications with child components having legacy style overrides involving margins.

## Testing Instructions

1. `npm run storybook:dev` and check that Flex, VStack, and HStack still work as expected when the `gap` and `wrap` props are set.
2. `npm run dev` and smoke test the editor UI.

There is a good chance of whitespace regressions where there are margin overrides involved. The only one I could find was a regression in CustomSelectControl, which I fixed at the root by removing margin overrides in InputControl.

## Screenshots

This cleanup also fixes a bug in InputControl when `labelPosition='bottom'`:

| Before | After |
|--------|-------|
|<img src="https://user-images.githubusercontent.com/555336/189200561-5dbcb348-7e38-477e-bfef-131aa039c541.png" alt="InputControl with bottom label, before" width="400">|<img src="https://user-images.githubusercontent.com/555336/189200571-68d933c0-a90c-4968-a5ff-3e92fccb0cb7.png" alt="InputControl with bottom label, after" width="400">|

Otherwise, there shouldn't be any visual changes.
